packages/hydrogen/src/cache/create-with-cache.test.ts

```diff
@@ -0,0 +1,116 @@
+import {describe, beforeEach, it, expect, vi} from 'vitest';
+import {type WithCache, createWithCache} from './create-with-cache';
+import {InMemoryCache} from './in-memory';
+import {getItemFromCache} from './sub-request';
+import {CacheNone, CacheShort} from './strategies';
+
+describe('createWithCache', () => {
+  const KEY = 'my-key';
+  const VALUE = 'my-value';
+  const actionFn = vi.fn(() => VALUE);
+  const waitUntil = vi.fn(() => {});
+  let cache: InMemoryCache;
```

Comment:
For now this test uses our InMemoryCache implementation. We could change this to eventually use real MiniOxygen cache directly but I think it's good for now.

Reply:
This is awesome! Thank you for doing this. One nitpicky thing, but maybe an extra test that just validates cache expiration?